mongo/src/main/java/com/redhat/lightblue/mongo/crud/BasicDocFinder.java

```diff
@@ -85,7 +85,7 @@ public long find(CRUDOperationContext ctx,
                 cursor.setReadPreference(readPreference);
             }
 
-            if (maxQueryTimeMS > 0) {
+            if (maxQueryTimeMS > 0&&ctx.isLimitQueryTime()) {
```

Comment:
I added that thinking with streaming, there is no need, but that will still be useful if you submit a long running query, so it is there just in case.